src/eva/vision/data/datasets/segmentation/lits.py

```diff
@@ -0,0 +1,178 @@
+
+    @override
+    def validate(self) -> None:
+        if len(self._volume_files) != len(self._segmentation_files):
+            raise ValueError(
+                "The number of volume files does not match the number of the segmentation ones."
+            )
+
+        _validators.check_dataset_integrity(
+            self,
+            length=self._expected_dataset_lengths.get(self._split, 0),
+            n_classes=2,
+            first_and_last_labels=("liver", "tumor"),
+        )
+
+    @override
+    def load_image(self, index: int) -> tv_tensors.Image:
+        sample_index, slice_index = self._indices[index]
+        volume_path = self._volume_files[sample_index]
+        image_array = io.read_nifti(volume_path, slice_index)
+        return tv_tensors.Image(image_array.transpose(2, 0, 1))
+
+    @override
+    def load_mask(self, index: int) -> tv_tensors.Mask:
+        sample_index, slice_index = self._indices[index]
+        segmentation_path = self._segmentation_files[sample_index]
+        semantic_labels = io.read_nifti(segmentation_path, slice_index)
+        return tv_tensors.Mask(semantic_labels.squeeze(), dtype=torch.int64)  # type: ignore[reportCallIssue]
+
+    @override
+    def load_metadata(self, index: int) -> Dict[str, Any]:
+        _, slice_index = self._indices[index]
+        return {"slice_index": slice_index}
+
+    @override
+    def __len__(self) -> int:
+        return len(self._indices)
+
+    def _get_number_of_slices_per_volume(self, sample_index: int) -> int:
+        """Returns the total amount of slices of a volume."""
+        file_path = self._volume_files[sample_index]
+        volume_shape = io.fetch_nifti_shape(file_path)
+        return volume_shape[-1]
+
+    @functools.cached_property
+    def _volume_files(self) -> List[str]:
+        files_pattern = os.path.join(self._root, "**", "volume-*.nii")
+        files = glob.glob(files_pattern, recursive=True)
+        return utils.numeric_sort(files)
+
+    @functools.cached_property
+    def _segmentation_files(self) -> List[str]:
+        files_pattern = os.path.join(self._root, "**", "segmentation-*.nii")
+        files = glob.glob(files_pattern, recursive=True)
+        return utils.numeric_sort(files)
+
+    def _create_indices(self) -> List[Tuple[int, int]]:
+        """Builds the dataset indices for the specified split.
+
+        Returns:
+            A list of tuples, where the first value indicates the
+            sample index which the second its corresponding slice
+            index.
+        """
+        indices = [
+            (sample_idx, slide_idx)
+            for sample_idx in self._get_split_indices()
+            for slide_idx in range(self._get_number_of_slices_per_volume(sample_idx))
+            if slide_idx % (self._sample_every_n_slices or 1) == 0
+        ]
+        return indices
+
+    def _get_split_indices(self) -> List[int]:
+        """Returns the samples indices that corresponding the dataset split and version."""
```

Comment:
-> """Returns the sample indices for the specified dataset split."""

Reply:
better yes